src/System.Text.Json/src/System/Text/Json/Serialization/ClassType.cs

```diff
@@ -9,11 +9,20 @@ namespace System.Text.Json
     /// </summary>
     internal enum ClassType
     {
-        Unknown = 0,        // typeof(object)
-        Object = 1,         // POCO or rich data type
-        Value = 2,          // Data type with single value
-        Enumerable = 3,     // IEnumerable
-        Dictionary = 4,     // IDictionary
-        ImmutableDictionary = 5, // Immutable Dictionary
+        // typeof(object)
+        Unknown = 0,
+        // POCO or rich data type
+        Object = 1,
+        // Data type with single value
+        Value = 2,
+        // IEnumerable
+        Enumerable = 3,
+        // IDictionary
+        Dictionary = 4,
+        // Is deserialized by passing a IDictionary to its constructor
+        // i.e. immutable dictionaries, Hashtable, SortedList,
+        IDictionaryConstructible = 5,
+        // KeyValuePair
+        KeyValuePair = 6,
```

Comment:
This is unfortunate to have to create another enum value for this. When the converter code goes in, I believe we can just write a converter for this as it will be able to handle multiple generic types `<TKey, TValue>`